oswin_tempest_plugin/tests/_mixins/migrate.py

```diff
@@ -43,3 +44,64 @@ def test_migration(self):
         server_tuple = self._create_server()
         self._migrate_server(server_tuple)
         self._check_server_connectivity(server_tuple)
+
+
+class _LiveMigrateMixin(object):
+    """Live migration mixin.
+
+    This mixin will add a Live migration test. It will perform the
+    following operations:
+
+    * Spawn instance.
+    * Live migrate the instance
+    * Check the server connectivity.
+    """
+
+    max_microversion = '2.24'
+
+    def _live_migrate_server(self, server_tuple, destination_host=None,
+                             state='ACTIVE', volume_backed=False):
+        kwargs = dict()
+        server = server_tuple.server
+        admin_server = self._get_server_as_admin(server)
+        current_host = admin_server['OS-EXT-SRV-ATTR:host']
+
+        block_migration = (CONF.compute_feature_enabled.
+                           block_migration_for_live_migration and
+                           not volume_backed)
+
+        kwargs['disk_over_commit'] = False
+
+        self.admin_servers_client.live_migrate_server(
+            server['id'],
+            host=destination_host,
+            block_migration=block_migration,
+            **kwargs)
+
+        waiters.wait_for_server_status(self.admin_servers_client, server['id'],
+                                       state)
+
+        admin_server = self._get_server_as_admin(server)
+        after_migration_host = admin_server['OS-EXT-SRV-ATTR:host']
+
+        migration_list = (self.admin_migrations_client.list_migrations()
+                          ['migrations'])
+
+        msg = ("Live Migration failed. Migrations list for Instance "
+               "%s: [" % server['id'])
+        for live_migration in migration_list:
+            if (live_migration['instance_uuid'] == server['id']):
+                msg += "\n%s" % live_migration
+        msg += "]"
+
+        if destination_host:
+            self.assertEqual(destination_host, after_migration_host, msg)
+        else:
+            self.assertNotEqual(current_host, after_migration_host, msg)
+
+    @testtools.skipUnless(CONF.compute.min_compute_nodes >= 2,
+                          'Expected at least 2 compute nodes.')
```

Comment:
there should be another config option, which basically specifies if the live_migration feature is enabled. we should check for that too.